frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -21,32 +27,196 @@ const AlertRouting: React.FC<AlertRoutingProps> = ({ config, secret }) => {
+    <TableRow id={index} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>{receiver.name}</TableData>
+      <TableData className={tableColumnClasses[1]}>{getIntegrationTypes(receiver)}</TableData>
+      <TableData className={tableColumnClasses[2]}>
+        {_.map(receiverRoutingLabels, (route, i) => (
+          <div key={i}>
+            <RoutingLabel labels={_.get(route, 'labels')} />
+          </div>
+        ))}
+      </TableData>
+    </TableRow>
+  );
+};
+ReceiverTableRow.displayName = 'ReceiverTableRow';
+
+const ReceiversTable: React.FC<ReceiverTableProps> = (props) => {
+  const { route, filterValue } = props;
+  const { receiver: defaultReceiver, routes } = route;
+
+  const routingLabels = getRoutinglabels(routes, {});
+  if (defaultReceiver) {
+    routingLabels.push({ receiver: defaultReceiver, labels: { default: 'all' } });
+  }
+  const EmptyMsg = () => <MsgBox title={`No Receivers match filter '${filterValue}'`} />;
+  return (
+    <Table
+      {...props}
+      aria-label="Receivers"
+      customData={routingLabels}
+      EmptyMsg={EmptyMsg}
+      Header={ReceiverTableHeader}
+      Row={ReceiverTableRow}
+      loaded={true}
+      defaultSortField="name"
+      virtualize
+    />
+  );
+};
+ReceiversTable.displayName = 'ReceiversTable';
+
+const ReceiversEmptyState: React.FC = () => (
+  <EmptyState variant={EmptyStateVariant.full}>
+    <Title size="lg">No Receivers Found</Title>
+    <EmptyStateBody>
+      Create a receiver to get OpenShift alerts through other services such as email or a chat
+      platform. The first receiver you create will become the default receiver and will
+      automatically receive all alerts from this cluster. Subsequent receivers can have specific
+      sets of alerts routed to them.
+    </EmptyStateBody>
+  </EmptyState>
+);
+
+const Receivers: React.FC<AlertManagerProps> = ({ config }) => {
+  const [receiverFilter, setReceiverFilter] = React.useState('');
+  let receivers = _.get(config, 'receivers', []);
+  if (receiverFilter) {
+    const filterStr = _.toLower(receiverFilter);
+    receivers = receivers.filter((receiver) => fuzzy(filterStr, _.toLower(receiver.name)));
+  }
+  return (
+    <div className="co-m-pane__body">
+      <SectionHeading text="Receivers" />
+      <div className="co-m-pane__filter-bar co-m-pane__filter-bar--alt">
+        <div className="co-m-pane__filter-bar-group co-m-pane__filter-bar-group--filter">
+          <TextFilter
+            defaultValue=""
+            label="Receivers by Name"
+            onChange={(e) => setReceiverFilter(e.target.value)}
+          />
+        </div>
+      </div>
+      {(!_.isEmpty(receivers) || (_.isEmpty(receivers) && receiverFilter)) && (
+        <ReceiversTable
+          filterValue={receiverFilter}
+          data={receivers}
+          route={_.get(config, ['route'])}
+        />
+      )}
+      {_.isEmpty(receivers) && !receiverFilter && <ReceiversEmptyState />}
```

Comment:
Is this the same logic?
```
{(_.isEmpty(receivers) && !receiverFilter)
  ? <ReceiversEmptyState />
  : <ReceiversTable
    filterValue={receiverFilter}
    data={receivers}
    route={_.get(config, 'route')}
  />}
```

Reply:
yes, that works & easier to read, thanks :-)